crates/hyperdrive-math/src/lp/math.rs

```diff
@@ -629,6 +1295,402 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn fuzz_test_calculate_distribute_excess_idle() -> Result<()> {
+        let chain = TestChain::new().await?;
+        let alice = chain.alice().await?;
+        let mock = chain.mock_lp_math();
+
+        // Fuzz the rust and solidity implementations against each other.
+        let mut rng = thread_rng();
+
+        for _ in 0..*FAST_FUZZ_RUNS {
+            // Generate random states.
+            let mut present_state = rng.gen::<State>();
+
+            // Make sure maturity times are in the future.
+            let current_block_timestamp = alice.now().await?;
+            present_state.info.long_average_maturity_time += current_block_timestamp;
+            present_state.info.short_average_maturity_time += current_block_timestamp;
+
+            // active_lp_total_supply and _withdrawal_shares_total_supply are just the token supplies.
+            // Neither are supplied in the PoolInfo so we need to make them here.  lp_total_supply is defined as:
+            // lp_total_supply = active_lp_total_supply + withdrawal_shares_total_supply - withdrawal_shares_ready_to_withdraw
+            // active_lp_total_supply = lp_total_supply - withdrawal_shares_total_supply + withdrawal_shares_ready_to_withdraw
+            // We clip withdrawal_shares_total_supply to ensure active_lp_total_supply doesn't underflow.
+            let active_lp_total_supply = present_state.lp_total_supply()
+                + present_state.withdrawal_shares_ready_to_withdraw();
```

Comment:
This should be `lp_total_supply + (withdrawal_share_total_supply - withdrawal_shares_ready_to_withdraw)`.

Reply:
oh yeah, this is happening two lines below.  i do it piecemeal to make sure i don't underflow when i subtract withdrawal_shares_ready_to_withdraw.